### ISSUE TITLE
Undefined cursor if no connection is found (v2.0.0-alpha.3)

### ISSUE THREAD

**Issue by kristjanmik** [neo4j/graphql#327]
**Describe the bug**
After the #282 merge, I'm getting the error: Cannot read property 'cursor' of undefined.
This only happens when no nodes are connected with a relationship, but the error is not raised if relationships and nodes are found.

**Type definitions**
```gql
type Query {
  activity(id: ID!): Activity
}

interface ActivityHasMediaProperties {
  timestamp: DateTime
}

type Activity {
  media: [Media]!
    @relationship(
      type: "HAS_MEDIA"
      properties: "ActivityHasMediaProperties"
      direction: OUT
    )
}
```

**To Reproduce**

**Querying for activity1 works. This is expected as at least one relationship exists.**
![Screenshot 2021-07-16 at 12 18 26](https://user-images.githubusercontent.com/911107/125946277-910c262d-be49-46d9-b1ad-984661007fd4.png)

```gql

query Activity {
  activity(id: "activity1") {
    id
    mediaConnection{
      edges{
        timestamp
      }
    }
  }
}
```

Results
```json
{
  "data": {
    "activity": {
      "id": "activity1",
      "mediaConnection": {
        "edges": [
          {
            "timestamp": "2021-04-14T00:00:00.000Z"
          }
        ]
      }
    }
  }
}
```

**Querying for activity3 results in the above error. This is not expected, should return an empty array.**
![Screenshot 2021-07-16 at 12 18 37](https://user-images.githubusercontent.com/911107/125946370-d2ab2050-fe00-4138-b042-dc1b7ccb035d.png)

```gql
query Activity {
  activity(id: "activity3") {
    id
    mediaConnection{
      edges{
        timestamp
      }
    }
  }
}
```

Results
```json
{
  "errors": [
    {
      "message": "Cannot read property 'cursor' of undefined",
      "locations": [
        {
          "line": 151,
          "column": 3
        }
      ],
      "path": [
        "activity"
      ],
      "extensions": {
        "code": "INTERNAL_SERVER_ERROR",
        "exception": {
          "stacktrace": [
            "Error: Cannot read property 'cursor' of undefined",
            "    at Model.<anonymous> (/...masked.../node_modules/@neo4j/graphql-ogm/src/classes/Model.ts:105:19)",
            "    at Generator.next (<anonymous>)",
            "    at fulfilled (/...masked.../node_modules/@neo4j/graphql-ogm/dist/classes/Model.js:23:58)",
            "    at processTicksAndRejections (internal/process/task_queues.js:93:5)"
          ]
        }
      }
    }
  ],
  "data": {
    "activity": null
  }
}
```

**Expected behavior**
Querying for activity 3 should return the following without errors.
```json
{
  "data": {
    "activity": {
      "id": "activity3",
      "mediaConnection": {
        "edges": [ ]
      }
    }
  }
}
```

**Comment by darrellwarde** [comment]
Brilliant, I'm looking at a few bugs around edges, cursors, totalCount and pageInfo now, so this should help in reproduction. Thanks!

**Comment by dmoree** [comment]
I am seeing this same error. Below is a reproduction based off the `neo-push` example.

Steps to reproduce:
1. `git clone https://github.com/dmoree/neo4j-graphql-issue-327.git`
2. `cd neo4j-graphql-issue-327 && yarn install && yarn seed && yarn start`
3. Run following `query` in GraphiQL at `http://localhost:4000/graphql` (or other as defined in `.env`)

```graphql
query PostsWithoutComments {
  posts {
    commentsConnection {
      edges {
        node {
          id
        }
      }
    }
  }
}
```
4. error

Graph: (basically reproducing `neo-push` except not creating any comments on any posts)
![Screen Shot 2021-07-16 at 7 35 04 PM](https://user-images.githubusercontent.com/10360334/126022714-c5a167c4-6cbd-4f19-963e-14f7449e4162.png)

**Comment by litewarp** [comment]
I'm 95% sure the issue is in the [pagination](https://github.com/neo4j/graphql/blob/2c757e3083035a3bf5d7b880bf20de9fb347d53d/packages/graphql/src/schema/pagination.ts#L56) helper when it calculates the returned pageInfo object. 

I believe you just need to change it so that the start and end cursors are null when the count is zero.

**Comment by dmoree** [comment]
@litewarp I'm not too familiar with it, but according to the Relay specification on Connection types:

> A “Connection Type” must contain a field called pageInfo. This field must return a non-null PageInfo object, as defined in the “PageInfo” section below.

https://relay.dev/graphql/connections.htm#sel-FAFFHDCAACE0Bnld

And then in the PageInfo section it states:

>  PageInfo must contain fields hasPreviousPage and hasNextPage, both of which return non-null booleans. It must also contain fields startCursor and endCursor, both of which return non-null opaque strings.

https://relay.dev/graphql/connections.htm#sel-EALFDDAAACJvB_iO

My question would be that in the situation with zero edges would you return a `PageInfo` object with a `startCursor` and `endCursor` with empty strings. Otherwise, what would you return in this case?

**Comment by litewarp** [comment]
Great question. So the `graphql-relay` package (which I've been following for reference) [returns null](https://github.com/graphql/graphql-relay-js/blob/17231860ee64431009c193be2e6ac444936a88ea/src/connection/arrayConnection.js#L103) for the end and start cursors when the edges array is empty. 

Thus, when there the total number of edges is zero, it will return a pageInfo object like the following:

```
pageInfo: {
  hasNextPage: false,
  hasPreviousPage: false,
  startCursor: null,
  endCursor: null
}
```

Also see PR #331

**Comment by dmoree** [comment]
That makes sense to me. I couldn't see how to return a non-null string for `startCursor` when you have zero edges. Thanks for the PR.

**Comment by darrellwarde** [comment]
The fix for this will be released in 2.0.0-alpha.4 shortly, please let us know if it works for you! 🙂
